### PR TITLE
Export ICountOptions and use Generics.

### PR DESCRIPTION
**Pull request by santiagodoldan**

### CODE REVIEW

**Comment by bilby91** [comment]
@RobinBuschmann Are you fine with this change ?

File: lib/models/Model.d.ts [hunk 1]
```diff
@@ -234,7 +234,7 @@ export declare abstract class Model<T> extends Hooks {
    *
    * If you provide an `include` option, the number of matching associations will be counted instead.
    */
-  static count(options?: ICountOptions): Promise<number>;
+  static count<T extends Model<T>>(options?: ICountOptions<T>): Promise<number>;
```
Comment (RobinBuschmann) [inline lib/models/Model.d.ts]: Can you add `this: (new () => T)` here as well? See `findAll` as an example: https://github.com/suttna/sequelize-typescript/blob/42240661d503d614781d1353f131a37db33300be/lib/models/Model.d.ts#L201

**Comment by RobinBuschmann** [comment]
@santiagodoldan Thanks for contributing @santiagodoldan @bilby91 There is one thing left - see the my comment :)

**Comment by santiagodoldan** [comment]
@RobinBuschmann done, thanks to you ☺️

**Pull request merged**